### PR TITLE
Add code owner policy

### PR DESCRIPTION
**Pull request by MrAlias**
Close https://github.com/open-telemetry/opentelemetry-go-contrib/issues/2834

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -9,6 +9,44 @@ Before you start please be sure to read through these contributing requirements
...
+
+### Requirements
+
+To become a Code Owner, you will need to meet the following requirements.
+
+1. You will need to be a [member of the OpenTelemetry organization] and maintain that membership.
+2. You need to have good working knowledge of the code you are sponsoring and any project that that code instruments or is based on.
+
+If you are not an existing member, this is not an imediate disqualification.
+You will need to engate with the OpenTelemetry community so you can achieve this membership in the process of becoming a Code Owner.
```
Comment (dmathieu) [inline CONTRIBUTING.md:27]:
```suggestion
You will need to engage with the OpenTelemetry community so you can achieve this membership in the process of becoming a Code Owner.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -9,6 +9,44 @@ Before you start please be sure to read through these contributing requirements
...
+
+It is best to have resolved an issue related to the module, contributed directly to the module, and/or review module PRs.
```
Comment (dmathieu) [inline CONTRIBUTING.md:29]:
```suggestion
It is best to have resolved an issue related to the module, contributed directly to the module, and/or reviewed module PRs.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -9,6 +9,44 @@ Before you start please be sure to read through these contributing requirements
...
+How much interaction with the module is required before becoming a Code Owner is up to the existing Code Owners.
+
+Code Ownership is ultimately up to the judgement of the existing Code Owners and Maintainers of this repository.
+Meeting the above requirements is not a guarantee to be granted Code Ownership.
+
+[member of the OpenTelemetry organization]: https://github.com/open-telemetry/community/blob/main/community-membership.md#member
+
+### Responsibilities
+
+As a Code Owner you will be responsible for the following.
```
Comment (dmathieu) [inline CONTRIBUTING.md:39]:
```suggestion
As a Code Owner you will be responsible for the following:
```

**Review by dashpole — APPROVED** [review]
Do we want to add anything about code owner removal?  E.g. if someone becomes inactive?

**Comment by dmathieu** [comment]
How about something along the lines of:

```
Code Owners are expected to remove their ownership if they cannot fulfill their responsibilities anymore.
Existing Code Owners may remove ownership of an owner if they are unable to demonstrate their investment.
```

**Comment by dashpole** [comment]
Maybe
```
Code Owners are expected to remove their ownership if they cannot fulfill their responsibilities anymore.
Maintainers or existing code owners may remove ownership of an owner if they are unable to demonstrate their investment.
```

**Comment by MrAlias** [comment]
> ```
> Existing Code Owners may remove ownership of an owner if they are unable to demonstrate their investment.
> ```

Do we want to quantify "unable to demonstrate their investment"?

Does this mean they are inactive for a month or two? Have they changed jobs and their focus isn't in the area any more?

**Comment by dmathieu** [comment]
Some instrumentations have less than monthly updates.
How would we say someone has been inactive for too long if it's the instrumentation which isn't getting activity?

We're being pretty clear in this doc that it's an oligarchy, and we'll accept whoever we want (which is fine). I think we can say the same for removing folks.

**Comment by MrAlias** [comment]
> Some instrumentations have less than monthly updates. How would we say someone has been inactive for too long if it's the instrumentation which isn't getting activity?
> 
> We're being pretty clear in this doc that it's an oligarchy, and we'll accept whoever we want (which is fine). I think we can say the same for removing folks.

I would like to avoid the situation where there are disagreeing personalities that become the reason to disqualify an owner.

Having explicit requirements for this dismissal helps prevent this.

**Comment by dashpole** [comment]
6 months seems like a generous amount of time.  We can always add "or at the discretion of maintainers" if we want to give ourselves flexibility.

**Comment by MrAlias** [comment]
Updated. PTAL

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -9,6 +9,54 @@ Before you start please be sure to read through these contributing requirements
...
+That responsibility includes maintaining the component, triaging and responding to issues, and reviewing pull requests.
+
+### Requirements
+
+To become a Code Owner, you will need to meet the following requirements.
+
+1. You will need to be a [member of the OpenTelemetry organization] and maintain that membership.
+2. You need to have good working knowledge of the code you are sponsoring and any project that that code instruments or is based on.
+
+If you are not an existing member, this is not an imediate disqualification.
```
Comment (dmathieu) [inline CONTRIBUTING.md:26]:
```suggestion
If you are not an existing member, this is not an immediate disqualification.
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -9,6 +9,54 @@ Before you start please be sure to read through these contributing requirements
...
+You will need to engage with the OpenTelemetry community so you can achieve this membership in the process of becoming a Code Owner.
+
+It is best to have resolved an issue related to the module, contributed directly to the module, and/or reviewed module PRs.
```
Comment (dmathieu) [inline CONTRIBUTING.md:29]:
```suggestion
It is best to have resolved at least an issue related to the module, contributed directly to the module, and/or reviewed module PRs.
```

**Review by dmathieu — APPROVED** [review]

**Review by dashpole — APPROVED** [review]

**Pull request merged**